### PR TITLE
[KEP-2371] Fixing bug with windows criStatsProvider sometimes not reporting AvailableBytes correctly

### PR DESCRIPTION
**Pull request by marosset**
#### What type of PR is this?
/kind bug

#### Does this PR introduce a user-facing change?

```release-note
Fixing issue where AvailableBytes sometimes does not report correctly on WindowsNodes when PodAndContainerStatsFromCRI feature is enabled.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/area kubelet
/sig node windows
/assign @jsturtevant @dashpole @haircommander

### CODE REVIEW

**Comment by marosset** [comment]
Fixes a possible issue in stats reporting on windows nodes when https://github.com/kubernetes/enhancements/issues/2371 is enabled

**Comment by jsturtevant** [comment]
> Fixes a possible issue in stats reporting on windows nodes when [kubernetes/enhancements#2371](https://github.com/kubernetes/enhancements/issues/2371) is enabled

I believe this is for the CRI only metrics feature gate which isn't enabled on our test runs so I don't think it will fix that but we can monitor it after it goes in.

/lgtm

**Comment by kannon92** [comment]
Could you add a release note if this is fixing a user facing bug?

**Comment by kannon92** [comment]
And since there already exists a unit test file for this area, could you add a test to catch issues like this?

**Comment by marosset** [comment]
Yea, I can add some new unit tests

**Comment by marosset** [comment]
> Could you add a release note if this is fixing a user facing bug?

The bug is only code that is still in alpha (the CRI-only stats work) so I think maybe it would be better to not call this out in the release notes?

**Comment by kannon92** [comment]
/retitle [KEP-2371] Fixing bug with windows criStatsProvider sometimes not reporting AvailableBytes correctly

**Comment by marosset** [comment]
> Yea, I can add some new unit tests

Added!

File: pkg/kubelet/stats/cri_stats_provider_windows_test.go [hunk 1]
```diff
@@ -426,3 +434,124 @@ func Test_criStatsProvider_listContainerNetworkStats(t *testing.T) {
...
+		Rootfs: &statsapi.FsStats{},
+		Logs: &statsapi.FsStats{
+			Time:       c0LogStats.Time,
+			UsedBytes:  toP(logStatsUsed),
+			InodesUsed: toP(logStatsInodesUsed),
+		},
+	}
+
+	if err != nil {
+		t.Errorf("makeWinContainerStats() error = %v, expected no error", err)
```
Comment (yujuhong) [inline pkg/kubelet/stats/cri_stats_provider_windows_test.go:538]: nit: why not just call t.Fatalf()?

Comment (marosset) [reply pkg/kubelet/stats/cri_stats_provider_windows_test.go]: updated - thanks!

**Comment by yujuhong** [comment]
Don't we need a release note since it's fixing a bug? (I have been away from the community for a while, so let me know if the rules have changed :) )

**Comment by yujuhong** [comment]
lgtm but needs the release note :)

**Comment by marosset** [comment]
> lgtm but needs the release note :)

I left it out because this code is only enabled when `PodAndContainerStatsFromCRI` enabled which is still in alpha but I added one now.

File: pkg/kubelet/stats/cri_stats_provider_windows_test.go [hunk 2]
```diff
@@ -426,3 +434,124 @@ func Test_criStatsProvider_listContainerNetworkStats(t *testing.T) {
...
+		Logs: &statsapi.FsStats{
+			Time:       c0LogStats.Time,
+			UsedBytes:  toP(logStatsUsed),
+			InodesUsed: toP(logStatsInodesUsed),
+		},
+	}
+
+	if err != nil {
+		t.Fatalf("makeWinContainerStats() error = %v, expected no error", err)
+		return
```
Comment (yujuhong) [inline pkg/kubelet/stats/cri_stats_provider_windows_test.go:539]: Don't need the `return` now that you use t.Fatalf

**Comment by MaryamTavakkoli** [comment]
Hello! 
Release Signal shadow here.
The code freeze is starting [02:00 UTC Wednesday 6th March 2024 / 18:00 PDT Tuesday 5th March 2024](https://everytimezone.com/s/5056bace) (less than a month from now), and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

As this PR is tagged for 1.30, is it still planned for this release?

**Comment by yujuhong** [comment]
@marosset could fix the nit? thanks.

**Comment by yujuhong** [comment]
/lgtm

No worries. Thanks!

**Pull request merged**